tests/src/Unit/Export/ExportManagerTest.php

```diff
@@ -183,6 +224,98 @@ private function branchSupportedOperation(
     $test_context['logger']
       ->error(Argument::any())
       ->shouldNotBeCalled();
+
+    // Continue on with the next set of tests.
+    $this->branchEmptyEntityMapping($test_context);
+    $this->branchSkipAction($test_context);
+    $this->branchUnsupportedAction($test_context);
+  }
+
+  /**
+   * Assert that we return if the entity mapping array is empty.
+   *
+   * @param array $test_context
+   *   The test case context.
+   */
+  private function branchEmptyEntityMapping(array $test_context) {
+    if ($test_context['event_entity_mapping']) {
+      return;
+    }
+
+    // @I Assert that the function doesn't continue on
+    //    type     : task
+    //    priority : normal
+    //    labels   : test
+  }
+
+  /**
+   * Assert that we return if the entity mapping array returned a SKIP action.
+   *
+   * @param array $test_context
+   *   The test case context.
+   */
+  private function branchSkipAction(array $test_context) {
+    $mapping = $test_context['event_entity_mapping'];
+    if (!$mapping) {
+      return;
+    }
+
+    if ($mapping && $mapping['action'] !== ManagerInterface::ACTION_SKIP) {
+      return;
+    }
+
+    // @I Assert that the function doesn't continue on
+    //    type     : task
+    //    priority : normal
+    //    labels   : test
+  }
+
+  /**
+   * Assert that we throw an exception on entity mapping with an illegal action.
+   *
+   * @param array $test_context
+   *   The test case context.
+   */
+  private function branchUnsupportedAction(array $test_context) {
+    $supported_actions = [
+      ManagerInterface::ACTION_SKIP,
+      ManagerInterface::ACTION_EXPORT,
+    ];
+
+    $mapping = $test_context['event_entity_mapping'];
+    if (!$mapping || in_array($mapping['action'], $supported_actions)) {
+      return;
+    }
+
+    // Unsupported actions will result in an exception.
+    $this->expectException(\RuntimeException::class);
```

Comment:
@krystalcode Even though we're expecting the exception here, this assertion doesn't pass and I get the exception error:

```
Exception : Unsupported entity mapping action "unsupported-action".
```

Reply:
Some help on this one ^^ @krystalcode

Reply:
We just needed to test the type as well in the `in_array` test. That's why it never got to the assertion because it always returned TRUE. So, changed it to this:

```php
    if (!$mapping || in_array($mapping['action'], $supported_actions, TRUE)) {
      return;
    }
```